### PR TITLE
[microTVM] Add Arduino CLI support to ci-qemu

### PR DESCRIPTION
**Pull request by guberti**
When merging Arduino support from #8493, we'd like to be able to add unit tests. Since no simulators exist for any of the boards we currently support, we will only be able to test project generation and compilation.

Specifically, this PR adds an `ubuntu_install_arduino.sh` script that installs `arduino-cli` via `wget` (as recommended) and then downloads the SDKs for two of the boards we support.

### CODE REVIEW

File: docker/install/ubuntu_install_arduino.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+export DEBIAN_FRONTEND=noninteractive
+apt-get update --fix-missing
+apt-get install -y ca-certificates
+
+# Install arduino-cli latest version
+wget -O - https://raw.githubusercontent.com/arduino/arduino-cli/master/install.sh | sh -s
+
+# Board manager URLs, separated by semicolons
+BOARD_MANAGER_URLS=https://github.com/sonydevworld/spresense-arduino-compatible/releases/download/generic/package_spresense_index.json
```
Comment (areusch) [inline docker/install/ubuntu_install_arduino.sh:31]: should we list these in the CLI config? https://arduino.github.io/arduino-cli/latest/getting-started/#adding-3rd-party-cores

then we don't have to worry about forgetting this option later on. (or do we want to not worry about this? we don't want the CI to depend on some network resource that could become unavailable)

File: docker/install/ubuntu_install_arduino.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -e
+set -u
+set -o pipefail
+
+export DEBIAN_FRONTEND=noninteractive
+apt-get update --fix-missing
```
Comment (areusch) [inline docker/install/ubuntu_install_arduino.sh:24]: why's this necessary?

**Review by areusch — COMMENTED** [review]
thanks @guberti , see some comments inline

File: docker/install/ubuntu_install_arduino.sh (hunk 2 above)
Comment (guberti) [reply docker/install/ubuntu_install_arduino.sh:24]: You're right, it isn't. Fixed.

File: docker/install/ubuntu_install_arduino.sh (hunk 1 above)
Comment (guberti) [reply docker/install/ubuntu_install_arduino.sh]: Tweaked a little to make it clearer what depends on what. We only need to specify the `--additional-urls` flag when we run `arduino-cli core install` right below - after that, further commands don't need the flag or a network connection.

**Review by mehrdadh — CHANGES_REQUESTED** [review]
@guberti thanks for this change! Can you explain your plan on testing Arduino on ci_qemu?
Maybe add information about the board that you are using and etc.

**Comment by guberti** [comment]
As discussed with Mehrdad, the Sony Spresense SDK (the Arduino board I used for development) has [a bug](https://github.com/sonydevworld/spresense/issues/200) that prevents TVM from compiling. The fix is available on the development branch and will be released in version 2.3.0, but that has not yet occurred.

Instead of using a workaround (such as installing from the development branch or cherry-picking the commit), IMO it's better to just test on a different type of board. Commit fe1d4dd replaces the Spresense with the Arduino Due, which is much more popular and works without needing a crazy hack.

**Review by mehrdadh — APPROVED** [review]
I built the image and tested it. LGTM!
Thanks!

**Review by areusch — APPROVED** [review]

**Comment by areusch** [comment]
thanks @guberti, the PR is now merged!

**Pull request merged**